src/content/docs/en/guides/backend/supabase.mdx

```diff
@@ -3,10 +3,806 @@ title: Supabase & Astro
+          DB URL: postgresql://postgres:postgres@localhost:54322/postgres
+      Studio URL: http://localhost:54323
+    Inbucket URL: http://localhost:54324
+      JWT secret: super-secret-jwt-token-with-at-least-32-characters-long
+        anon key: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZS1kZW1vIiwicm9sZSI6ImFub24iLCJleHAiOjE5ODM4MTI5OTZ9.CRXP1A7WOeoJeXxjNni43kdQwgnWNReilDMblYTn_I0
+service_role key: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZS1kZW1vIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImV4cCI6MTk4MzgxMjk5Nn0.EGIM96RAZx35lJzdJsyH-qQwv8Hdp7fsn3W0YpN81IU
+```
+
+To add Supabase credentials to Astro, create an `.env` file in the root of the project with the following:
+
+```ini
+# .env
+SUPABASE_URL=SUPABASE_URL
+SUPABASE_ANON_KEY=SUPABASE_ANON_KEY
+SUPABASE_JWT_SECRET=SUPABASE_JWT_SECRET
+```
+
+The Project structure should now include this file:
+
+<FileTree title="Project Structure">
+- ...
+- supabase/
+- .env
+- ...
+</FileTree>
+
+Now, these environment variables are available for use in the project.
+
+To have IntelliSense for the Supabase environment variables, edit or create the file `env.d.ts` in `src/` directory and configure the types:
+
+```ts title="src/env.d.ts"
+interface ImportMetaEnv {
+  readonly SUPABASE_URL: string;
+  readonly SUPABASE_ANON_KEY: string;
+  readonly SUPABASE_JWT_SECRET: string;
+}
+
+interface ImportMeta {
+  readonly env: ImportMetaEnv;
+}
+```
+
+:::tip
+Read more about [environment variables](/en/guides/environment-variables/) and `.env` files in Astro.
+:::
+
+The project structure should include these files:
+<FileTree title="Project Structure">
+- ...
+- supabase/
+- src/
+  - env.d.ts
+- .env
+- ...
+</FileTree>
+
+### Install Dependencies
+
+To connect Astro with Supabase, install the following package using the command below:
+
+- `@supabase/supabase-js`
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+    ```shell
+    npm install @supabase/supabase-js
+    ```
+  </Fragment>
+  <Fragment slot="pnpm">
+    ```shell
+    pnpm install @supabase/supabase-js
+    ```
+  </Fragment>
+  <Fragment slot="yarn">
+    ```shell
+    yarn add @supabase/supabase-js
+    ```
+  </Fragment>
+</PackageManagerTabs>
+
+### Create Profile Table
+
+1. Go to [Studio URL](http://localhost:54323)
+2. Click on SQL Editor on the left sidebar
+3. Choose `User Management Starter` from `Quick start` in the bottom of the page
+4. Click `RUN`
+
+### Create Supabase Client
+
+1. Create a directory named `lib` in the `src/` directory.
+2. Create a file named `supabaseClient.ts` in the `src/lib/` directory.
+
+In `supabaseClient.ts` add the following code to create the supabase client.
+
+```ts title="src/lib/supabaseClient.ts"
+import { createClient } from '@supabase/supabase-js'
+
+export const supabase = createClient(
+  `${import.meta.env.SUPABASE_URL}`,
+  `${import.meta.env.SUPABASE_ANON_KEY}`,
+  {         
+    auth: {
+      persistSession: false,
```

Comment:
Since authentication will be handled server side we should enable `pkce`

Reply:
Exactly! I will fix it soon.